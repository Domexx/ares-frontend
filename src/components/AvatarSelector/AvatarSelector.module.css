/* === === === === AvatarSelector === === === === */

.AvatarSelector {
  margin-top: 0;
}

.AvatarSelector-logo {
  display: flex;
  grid-column: 7 / span 2;
  justify-content: center;
  margin-top: 30px;
}

.AvatarSelector-title {
  display: flex;
  grid-row: 2;
  grid-column: 5 / span 6;
  align-content: center;
  justify-content: center;
  font-size: 50px;
}

.AvatarSelector-list {
  display: flex;
  grid-row: 3;
  grid-column: 4 / span 8;
  justify-content: space-between;
}

.AvatarSelector-button {
  display: flex;
  grid-row: 4;
  grid-column: 7 / span 2;
  gap: 20px;
  align-items: center;
  justify-content: center;
  margin-top: 30px;
}

.AvatarSelector-button svg path {
  fill: var(--clr-dark-500);
}

@media (--laptop) {
  .AvatarSelector-button {
    grid-column: 6 / span 4;
  }
}

@media (--tablet-landscape) {
  .AvatarSelector-logo {
    grid-column: 4 / span 10;
  }

  .AvatarSelector-title {
    grid-column: 3 / span 12;
    font-size: 40px;
  }

  .AvatarSelector-list {
    grid-column: 4 / span 10;
  }

  .AvatarSelector-button {
    grid-column: 4 / span 10;
  }
}

@media (--phone) {
  .AvatarSelector-logo {
    grid-column: 4 / span 10;
  }

  .AvatarSelector-title {
    grid-column: 3 / span 12;
    font-size: 40px;
  }

  .AvatarSelector-list {
    grid-column: 4 / span 10;
  }

  .AvatarSelector-button {
    grid-column: 4 / span 10;
  }
}
