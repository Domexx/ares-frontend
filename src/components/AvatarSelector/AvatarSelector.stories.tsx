import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import AvatarSelectorComponent, { Props } from './AvatarSelector';

const Component = {
  title: 'Components/Avatar Selector',
  component: AvatarSelectorComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof AvatarSelectorComponent>> = (args) => {
  return <AvatarSelectorComponent {...args} />;
};

export const AvatarSelector = Template.bind({});
export default Component;

// https://storybook.js.org/docs/react/essentials/controls
AvatarSelector.args = {
  title: 'Cool Avatars',
  avatars: [
    {
      look: 'hd-180-1.ch-255-66.lg-280-110.sh-305-62.ha-1012-110.hr-828-61&action=std&gesture=std&direction=2&head_direction=3&size=n',
      name: 'Raizer',
      motto: 'im Raizer',
    },
    {
      look: 'hd-180-1.ch-255-66.lg-280-110.sh-305-62.ha-1012-110.hr-828-61&action=std&gesture=std&direction=2&head_direction=3&size=n',
      name: 'Eevee',
      motto: 'im Katzen',
    },
    {
      look: 'hd-180-1.ch-255-66.lg-280-110.sh-305-62.ha-1012-110.hr-828-61&action=std&gesture=std&direction=2&head_direction=3&size=n',
      name: 'Eevee',
      motto: 'im Katzen',
    },
  ],
} as Props;
