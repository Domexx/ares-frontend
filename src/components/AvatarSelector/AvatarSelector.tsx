import React from 'react';
import Button from '../../elements/Button';
import Logo from '../../elements/Logo';
import Avatar, { Props as AvatarProps } from './Avatar/Avatar';
import s from './AvatarSelector.module.css';

import { ReactComponent as CreateIcon } from '../../assets/static/images/Avatar-Selection/CreateIcon.svg';

export type Props = {
  title: string;
  avatars: AvatarProps[];
};

const AvatarSelector: React.FC<Props> = ({ title, avatars }) => {
  const avatarList = avatars.map((avatar, index) => {
    // eslint-disable-next-line react/no-array-index-key
    return <Avatar look={avatar.look} name={avatar.name} motto={avatar.motto} key={index} />;
  });

  return (
    <div className={`${s['AvatarSelector']} u-grid`}>
      <div className={s['AvatarSelector-logo']}>
        <Logo />
      </div>
      <h1 className={s['AvatarSelector-title']}>{title}</h1>
      <div className={s['AvatarSelector-list']}>{avatarList}</div>
      <Button className={s['AvatarSelector-button']}>
        Create new Character <CreateIcon />
      </Button>
    </div>
  );
};

export default AvatarSelector;
