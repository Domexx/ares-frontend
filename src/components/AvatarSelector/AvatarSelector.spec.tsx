import React from 'react';

import { render } from '@testing-library/react';

import AvatarSelector from './AvatarSelector';

describe('Avatar Selector Component', () => {
  it('should render properly', () => {
    const { container } = render(
      <AvatarSelector
        title="Avatar Selector Test Title"
        avatars={[
          {
            look: 'look',
            name: 'Testname',
            motto: 'Testmotto',
          },
        ]}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
