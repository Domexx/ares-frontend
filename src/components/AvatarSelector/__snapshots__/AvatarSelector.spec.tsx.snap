// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Selector Component should render properly 1`] = `
<div
  class="AvatarSelector u-grid"
>
  <div
    class="AvatarSelector-logo"
  />
  <h1
    class="AvatarSelector-title"
  >
    Avatar Selector Test Title
  </h1>
  <div
    class="AvatarSelector-list"
  >
    <div
      class="Avatar"
    >
      <div
        class="Avatar-figure"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27200%27%20height=%27300%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Avatar"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27220%27%20height=%27100%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Podium"
            class="Avatar-figurePodium"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <span
        class="Avatar-name"
      >
        Testname
      </span>
      <span
        class="Avatar-motto"
      >
        "
        Testmotto
        "
      </span>
    </div>
  </div>
</div>
`;
